worker.py

```
@@ -7,28 +8,44 @@
 
 
 def tally_votes(max_phrase_length=1200):
+    #if we have a word with equal vote value, it will take the first one
+    votedwords = r.hgetall('current_words')
+    highestvote = []
+    highestvote.append('')
+    highestvote.append(0)
+    for y in votedwords.items():
+        if int(y[1]) > highestvote[1]:
+            highestvote[0] = y[0]
+            highestvote[1] = y[1]
+    nbWordPhrase = r.hlen('current_phrase')
+    #add the winner word to the bucket current_phrase and delete
+    #the five word + vote in the bucket current_words
+    r.hset('current_phrase', highestvote[0], int(nbWordPhrase))
+    r.delete('current_words')
+    #find the number of letter in our phrase bucket, if len > 1200
+    #we remove the letter exceding 1200
+    phrasewords = r.getall('current_phrase')
+    lenwords = int(r.hlen('current_phrase')-1)
+    for words in phrasewords:
+        lenwords = lenwords + len(words)
+    if lenwords > 1200:
+        return 'phrase too long'
```

Here you'll want to use an Exception. Instead of returning a string use: raise Exception("Phrase too long")